source/manual/alerts/email-alert-api-unprocessed-work.html.md

```diff
@@ -0,0 +1,57 @@
+
+This alert indicates that Email Alert API has work that has not been processed in the generous amount of time we expect it to have been - see [here](https://github.com/alphagov/email-alert-api/tree/master/app/workers/metrics_collection_worker) for the timings for each alert.
+
+* **[`unprocessed content changes`](https://github.com/alphagov/email-alert-api/blob/master/app/workers/process_content_change_worker.rb)**. This means we're not generating emails for subscribers with "immediate" frequency subscriptions in response to [a change in some content] on GOV.UK.
+
+* **[`unprocessed messages`](https://github.com/alphagov/email-alert-api/blob/master/app/workers/process_message_worker.rb)**. This means we're not generating emails for subscribers with "immediate" frequency subscriptions in response to [a custom message].
+
+* **`incomplete digest runs`**. This could be due to a failure in any of three workers:
+
+   * [\[Daily/Weekly\]DigestInitiatorWorker](https://github.com/alphagov/email-alert-api/blob/a656389b1abdd46226ca37c1682c318f1c2eafee/app/workers/daily_digest_initiator_worker.rb) generates a DigestRunSubscriber work item for each subscriber.
+   * [DigestEmailGenerationWorker](https://github.com/alphagov/email-alert-api/blob/a656389b1abdd46226ca37c1682c318f1c2eafee/app/workers/digest_email_generation_worker.rb) does the work of generating the digest email for a specific subscriber.
+   * [DigestRunCompletionMarkerWorker](https://github.com/alphagov/email-alert-api/blob/a656389b1abdd46226ca37c1682c318f1c2eafee/app/workers/digest_run_completion_marker_worker.rb) periodically scans all the work items to see if the run is complete.
+
+Things to check:
+
+* Check [Sentry] for errors.
+
+* Check the [Sidekiq dashboard] for worker failures.
+
+* Check [Kibana] for errors - use ```@fields.worker: <worker class>``` for the query.
+
+* Manually retry any pending jobs from a console: `Sidekiq::RetrySet.new.retry_all`.
+
+* Check the [Email Alert API Technical dashboard] for performance issues.
+
+If all else fails, you can try running the work manually from a console. [The automatic recovery worker](https://github.com/alphagov/email-alert-api/blob/2f3931ac1ca25fe8c79b2405af98d1de55e1d47b/app/workers/recover_lost_jobs_worker/unprocessed_check.rb#L13) code is a good example of how to do this, but you will need to use `new.perform` instead of `perform_async`.
+
+> A digest run may be "complete" - all work items generated, all work items processed - but not marked as such. In this case, you will need to use slightly different commands to investigate the incomplete run:
+>
+> ```ruby
+> # find which digests are "incomplete"
+> DigestRun.where("created_at < ?", 1.hour.ago).where(completed_at: nil)
+>
+> # try manually marking it as complete
+> DigestRunCompletionMarkerWorker.new.perform
+> ```
```

Comment:
I'm not too sure how helpful this aside is - as this is just following the same step as given above to try `new.perform`. 

I'm figuring that you might instead be wanting to reference that `DigestRunCompletionMarkerWorker.new.perform` can successfully perform but not clear the alert as there could be DigestRunSubscribers that aren't processed? 

In general I don't think it's actually needed as all of the workers have nuances to learn if you're trying to debug the alert and we're not covering all of them.

Reply:
The re-running part is indeed the same. However, the previous paragraph was specific to manually trying jobs that are covered as part of automatic recovery, which doesn't include this particular worker.

The intro paragraph explains that the only scenario where this worker is important is when the marking itself is broken for some reason (but the work has all been done). I thought it helpful to indicate the code to investigate further.

> I'm figuring that you might instead be wanting to reference that DigestRunCompletionMarkerWorker.new.perform can successfully perform but not clear the alert as there could be DigestRunSubscribers that aren't processed?

That should not be the case, as the preceding guidance is inviting the reader to investigate this, since it's covered by the recovery worker.

Perhaps the distinction is too subtle?

Reply:
Ah sure, that makes sense. Hopefully we're getting rid of the thing soon anyway and can then remove the info.